### ISSUE TITLE
Caching of FilterSet cause problems with ForeignKey in v0.10.2

### ISSUE THREAD

**Issue by dzwiedziu-nkg** [philipn/django-rest-framework-filters#232]
In v0.10.2 of `filterset.py` you have:
```
        # attempt to retrieve related filterset subset from the cache
        key = cls.cache_key(filter_names)
        subset_class = cls.cache_get(key)

        # if no cached subset, then derive base_filters and create new subset
        if subset_class is not None:
            return subset_class
```

When I have:
```
class ParentOne(models.Model):
    ...

class ParentTwo(models.Model):
    ...

class ChildOne(models.Model):
    parent = models.ForeignKey(ParentOne)
    ...

class ChildTwo(models.Model):
    parent = models.ForeignKey(ParentTwo)
    ...
```

And `views.py`:
```
class ChildOne(viewsets.ModelViewSet):
    queryset = ChildOne.objects.all()
    filter_fields = {'parent': ['exact']}
    ...

class ChildTwo(viewsets.ModelViewSet):
    queryset = ChildTwo.objects.all()
    filter_fields = {'parent': ['exact']}
    ...
```

Then when I make `GET /api/child_one/?parent=3` and after it I make `GET /api/child_two/?parent=10` then I get `ValueError: Cannot query "ParentOne(...)": Must be "ParentTwo" instance.` because botch subsets in cache have the same name. When I comment caching of subsets its works.

**Comment by rpkilby** [comment]
Hi @dzwiedziu-nkg. In this case, you're not taking advantage of the cross-relationship filtering, so I would consider reverting to the standard `DjangoFilterBackend` for these views. eg, 


```python
class ChildTwo(viewsets.ModelViewSet):
    queryset = ChildTwo.objects.all()
    filter_fields = {'parent': ['exact']}
    filter_backends = ['django_filters.rest_framework.DjangoFilterBackend', ...]
    ...
```

Alternatively, you could create a `filter_class` with differing names so the caching can distinguish the two. 

```python
class ChildTwoFilterSet(rest_framework_filters.FilterSet):
    class Meta:
        model = ChildTwo
        fields = {'parent': ['exact']}


class ChildTwo(viewsets.ModelViewSet):
    queryset = ChildTwo.objects.all()
    filter_class = ChildTwoFilterSet
    ...
```


Either way, the caching will no longer be necessary, and has been removed in the next version of the package. That said, we should probably have a test that `filter_fields` is compatible with our backend.

**Comment by dzwiedziu-nkg** [comment]
Ok :) When you publish new version?

**Comment by rpkilby** [comment]
It will be a while. Development is dependent on some upstream django-filter features, and django-filter won't update until DRF 3.9 is released.

**Comment by rpkilby** [comment]
Hi @dzwiedziu-nkg. I'm closing this issue, as this is fixed on the master branch, and the 1.0 release should be ready very soon.
